packages/grid/x-data-grid/src/hooks/features/rows/useGridRows.ts

```diff
@@ -142,12 +143,24 @@ export const useGridRows = (
 
   const logger = useGridLogger(apiRef, 'useGridRows');
   const rowsCache = React.useRef(apiRef.current.state.rowsCache); // To avoid listing rowsCache as useEffect dep
+  const currentPage = useCurrentPageRows(apiRef, props);
 
   const getRow = React.useCallback<GridRowApi['getRow']>(
     (id) => gridRowsLookupSelector(apiRef)[id] ?? null,
     [apiRef],
   );
 
+  const lookup = React.useMemo(
+    () =>
+      currentPage.rows.reduce((acc, { id }, index) => {
```

Comment:
```suggestion
      currentPage.rows.reduce<Record<GridRowId, number>>((acc, { id }, index) => {
```

This should also work by the way